### PR TITLE
4909 remove deprecated args since v0.6 and v0.7

### LINKED ISSUE #4909: remove APIs that are marked deprecated since v0.6 and v0.7?
**Is your feature request related to a problem? Please describe.**
some APIs such as
https://github.com/Project-MONAI/MONAI/blob/535c6f1fc8767c09a21dba2464ca687f23c85154/monai/networks/blocks/convolutions.py#L101

https://github.com/Project-MONAI/MONAI/blob/535c6f1fc8767c09a21dba2464ca687f23c85154/monai/transforms/post/dictionary.py#L137-L154
are deprecated since version 0.6 and 0.7, they (and their corresonding tests) could be removed.

### PR DESCRIPTION
**Pull request by wyli**
Fixes #4909

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by KumoLiu — APPROVED** [review]
Looks good to me, thanks!

File: monai/networks/blocks/convolutions.py [hunk 1]
```diff
@@ -119,10 +112,9 @@ def __init__(
         is_transposed: bool = False,
         padding: Optional[Union[Sequence[int], int]] = None,
         output_padding: Optional[Union[Sequence[int], int]] = None,
-        dimensions: Optional[int] = None,
     ) -> None:
         super().__init__()
-        self.dimensions = spatial_dims if dimensions is None else dimensions
+        self.dimensions = spatial_dims
```
Comment (ericspod) [inline monai/networks/blocks/convolutions.py:117]: Should we change `self.dimensions` to `self.spatial_dims` as well or are there too many problems with that?

**Pull request merged**